app/gui/src/dashboard/components/dashboard/column/LabelsColumn.tsx

```diff
@@ -42,6 +50,24 @@ export default function LabelsColumn(props: column.AssetColumnProps) {
     category.type !== 'trash' &&
     (self?.permission === permissions.PermissionAction.own ||
       self?.permission === permissions.PermissionAction.admin)
+  const temporarilyAddedLabels = useStore(driveStore, ({ labelsDragPayload, dragTargetAssetId }) =>
+    (
+      showDraggedLabelsFallback ? labelsDragPayload?.typeWhenAppliedToSelection === 'add'
+      : item.id === dragTargetAssetId
+    ) ?
+      labelsDragPayload?.labels ?? EMPTY_ARRAY
+    : EMPTY_ARRAY,
+  )
```

Comment:
Can we simplify this one? Can't understand what's going on here.
My proposal:
```ts
  const temporarilyAddedLabels = useStore(
    driveStore,
    ({ labelsDragPayload, dragTargetAssetId }) => {
      if (showDraggedLabelsFallback && labelsDragPayload?.typeWhenAppliedToSelection === 'add') {
        return labelsDragPayload.labels
      }

      if (item.id === dragTargetAssetId) {
        return labelsDragPayload?.labels
      }

      return []
    },
  )
```

Reply:
Not sure I understood the fuction correctly, but I think I showed up the overall idea